backend/stakepoold/server.go

```diff
@@ -367,36 +427,158 @@ func (ctx *appContext) vote(wg *sync.WaitGroup, blockHash *chainhash.Hash, block
 
 	// Ask wallet to transmit raw transaction.
 	startSend := time.Now()
-	_, w.err = ctx.nodeConnection.SendRawTransaction(newTx, false)
+	tx, err := ctx.nodeConnection.SendRawTransaction(newTx, false)
+	if err != nil {
+		w.err = err
+	} else {
+		w.txid = tx
+	}
 	w.sendDuration = time.Since(startSend)
 }
 
+func (ctx *appContext) processNewTickets(nt NewTicketsForBlock) {
+	start := time.Now()
+
+	// We use pointer because it is the fastest accessor.
+	newtickets := make([]*ticketMetadata, 0, len(nt.newTickets))
+
+	var wg sync.WaitGroup // wait group for go routine exits
+
+	ctx.wgVoting.Wait() // wait for voting to finish
+
+	ctx.RLock()
+	for _, tickethash := range nt.newTickets {
+		n := &ticketMetadata{
+			ticket: tickethash,
+		}
+		newtickets = append(newtickets, n)
+
+		wg.Add(1)
+		go ctx.getticket(&wg, n)
+	}
+	ctx.RUnlock()
+
+	wg.Wait()
+
+	addtickets := make(map[chainhash.Hash]string)
+
+	for _, n := range newtickets {
+		if n.err != nil || n.msa == "" {
+			// most likely can't look up the transaction because it's
+			// not in our wallet because it doesn't belong to us
+			continue
+		}
+
+		addtickets[*n.ticket] = n.msa
+	}
+
+	ticketCountNew := 0
+	ticketCountOld := 0
+
+	ctx.Lock()
+	ticketCountOld = len(ctx.ticketsMSA)
+	for ticket, msa := range addtickets {
+		ctx.ticketsMSA[ticket] = msa
+	}
+	ticketCountNew = len(ctx.ticketsMSA)
+	ctx.Unlock()
+
+	// Log ticket information outside of the handler.
+	go func() {
+		for ticket, msa := range addtickets {
+			log.Infof("added new ticket %v msa %v", ticket, msa)
+		}
+
+		log.Infof("processNewTickets: height %v block %v duration %v "+
+			"newtickets %v ticketCountOld %v ticketCountNew %v", nt.blockHeight,
+			nt.blockHash, time.Since(start), len(addtickets), ticketCountOld,
+			ticketCountNew)
+	}()
+}
+
+func (ctx *appContext) processSpentMissedTickets(smt SpentMissedTicketsForBlock) {
+	start := time.Now()
+
+	// We use pointer because it is the fastest accessor.
+	smtickets := make([]*ticketMetadata, 0, len(smt.smTickets))
+
+	var wg sync.WaitGroup // wait group for go routine exits
+
+	ctx.wgVoting.Wait() // wait for voting to finish
+
+	ctx.RLock()
+	for ticket, spent := range smt.smTickets {
```

Comment:
The smTickets field still bugs me, being a `map[*chainhash.Hash]bool`.   It works as long as the only thing the map is used for is ranging over it to get the key-value pairs, but I'm afraid it might be used to lookup a hash, which won't work.